File: Lab3Java/app/src/main/java/pe/edu/upc/lab3java/AgendaActivity.java
package pe.edu.upc.lab3java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import java.util.ArrayList;
import java.util.HashMap;

public class AgendaActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_agenda);

        String[] from = new String[] { "Time", "Name", "Desc" };
        int[] to = new int[] { R.id.hora, R.id.nombre, R.id.desc1 };

        ArrayList<String[]> lista = new ArrayList<String[]>();

        String[] evento1 = { "11:30", "Reunión de coordinación", "Lugar: Auditorio", "1" };
        String[] evento2 = { "12:00", "Almuerzo", "Lugar: Comedor central", "2" };
        String[] evento3 = { "14:00", "Tiro al dardo", "Lugar: Oficina", "3" };

        lista.add(evento1);
        lista.add(evento2);
        lista.add(evento3);

        ArrayList<HashMap<String, String>> eventos = new ArrayList<HashMap<String, String>>();

        for (String[] evento : lista) {
            HashMap<String, String> datosEvento = new HashMap<String, String>();

            datosEvento.put("Time", evento[0]);
            datosEvento.put("Name", evento[1]);
            datosEvento.put("Desc", evento[2]);
            datosEvento.put("id", evento[3]);

            eventos.add(datosEvento);
        }

        SimpleAdapter listadoAdapter = new SimpleAdapter(this, eventos, R.layout.activity_agenda_fila, from, to);

        ListView lv = (ListView) findViewById(R.id.lista1);
        lv.setAdapter(listadoAdapter);

    }
}

File: Lab3Kotlin/app/src/main/java/pe/edu/upc/lab3kotlin/AgendaActivity.kt
package pe.edu.upc.lab3kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ListView
import android.widget.SimpleAdapter

class AgendaActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_agenda)

        val from = arrayOf("Time", "Name", "Desc")
        val to = intArrayOf(R.id.hora, R.id.nombre, R.id.desc1)

        val lista = ArrayList<Array<String>>()

        val evento1 = arrayOf("11:30", "Reunión de coordinación", "Lugar: Auditorio", "1")
        val evento2 = arrayOf("12:00", "Almuerzo", "Lugar: Comedor central", "2")
        val evento3 = arrayOf("14:00", "Tiro al blanco", "Lugar: Oficina", "3")

        lista.add(evento1);
        lista.add(evento2);
        lista.add(evento3);

        val eventos = ArrayList<HashMap<String, String>>()

        for (evento in lista) {
            val datosEvento = HashMap<String, String>()
            datosEvento["Time"] = evento[0]
            datosEvento["Name"] = evento[1]
            datosEvento["Desc"] = evento[2]
            datosEvento["id"] = evento[3]
            eventos.add(datosEvento)
        }

        val listadoAdapter = SimpleAdapter(this, eventos, R.layout.agenda_fila, from, to)
        val lv = findViewById<ListView>(R.id.lista1)
        lv.setAdapter(listadoAdapter)
    }
}